src/Components/PresentationalComponents/Filters/CustomFilters/CheckboxCustomFilter.js

```diff
@@ -0,0 +1,59 @@
+import React, { useState } from 'react';
+import propTypes from 'prop-types';
+import { Select, SelectOption } from '@patternfly/react-core';
+
+const CheckboxCustomFilter = ({ filterData, setFilterData, selectProps, options, filterName, filterId }) => {
+    const [isOpen, setOpen] = useState(false);
+
+    const handleOnCheckboxChange = (filterId, optionName) => {
+        const optionId = options.find(item => item.label === optionName).value;
+        if (!filterData[filterId].includes(optionId)) { // uncheck
+            setFilterData({ ...filterData, [filterId]: [...filterData[filterId], optionId] });
+        }
+        else { // check
+            let index = filterData[filterId].indexOf(optionId);
+            let temp = filterData[filterId].slice();
+            temp.splice(index, 1);
+            setFilterData({ ...filterData, [filterId]: temp });
+        }
+    };
+
+    return (
+        <Select
+            variant="checkbox"
+            aria-label="Select Input"
+            onToggle={() => setOpen(!isOpen)}
+            isOpen={isOpen}
+            placeholderText={`${filterName}: ${filterData[filterId].length === 0
```

Comment:
it would be great and cleaner if we could have a simple function that generates the `placeholderText`